nistats/reporting/glm_reporter.py

```diff
@@ -0,0 +1,913 @@
+                'contrast_name': contrast_name,
+                }
+            contrast_text_ = contrast_text_.safe_substitute(
+                    contrasts_for_subsitution
+                    )
+            all_contrasts_plots[contrast_name] = contrast_text_
+    return all_contrasts_plots
+
+
+def _make_headings(contrasts, title, model):
+    """ Creates report page title, heading & sub-heading
+     using title text or contrast names.
+    Accepts contrasts and user supplied title string or
+    contrasts and user supplied 3 element list or tuple.
+    
+    If title is not in (None, 'auto'),
+    page title == heading,
+    model type == sub-heading
+    
+    Parameters
+    ----------
+    contrasts: Dict[str, np.array or str]
+        Contrast information, as a dict in the form
+            {'contrast_title_1': contrast_info_1/title_1, ...}
+        Contrast titles are used in page title and secondary heading
+        if `title` is not 'auto' or None.
+    
+    title: String or List/Tuple with 3 elements
+        User supplied text for HTML Page title and primary heading.
+        Or 3 element List/Tuple for Title Heading, sub-heading resp.
+        Overrides title auto-generation.
+    
+    model: FirstLevelModel or SecondLevelModel
+        The model, passed in to determine its type
+        to be used in page title & headings.
+    Returns
+    -------
+    (HTML page title, heading, sub-heading): Tuple[str, str, str]
+        If title is user-supplied, then subheading is empty string.
+    """
+    if isinstance(title, (tuple, list)) and len(title) == 3:
+        return title
+
+    if type(model) == nistats.first_level_model.FirstLevelModel:
+        model_type = 'First Level Model'
+    elif type(model) == nistats.second_level_model.SecondLevelModel:
+        model_type = 'Second Level Model'
+    
+    if title not in ('auto', None):
+        return title, title, model_type
+    else:
+        if isinstance(contrasts, str):
+            contrasts_text = contrasts
+        else:
+            contrasts_names = sorted(contrasts)
+            contrasts_text = ', '.join(contrasts_names)
+        page_title = 'Report: {} for {}'.format(model_type, contrasts_text)
+        page_heading_1 = 'Statistical Report for {}'.format(model_type)
+        page_heading_2 = 'Contrasts: {}'.format(contrasts_text)
+        return page_title, page_heading_1, page_heading_2
+
+
+def _model_attributes_to_dataframe(model):
+    """ Returns an HTML table with pertinent model attributes & information.
+    
+    Parameters
+    ----------
+    model: FirstLevelModel or SecondLevelModel object.
+    
+    Returns
+    -------
+    HTML Table: String
+        HTML table with the pertinent attributes of the model.
+    """
+    selected_attributes = [
+        'subject_label',
+        'mask_img',
+        'drift_model',
+        'hrf_model',
+        'standardize',
+        'noise_model',
+        'min_onset',
```

Comment:
if we keep 'min_onset' then we should also keep 'oversampling' but I'm in favor of not keeping them (too low-level).